### PR TITLE
[WIP] feat/auth

### PR DESCRIPTION
**Pull request by sunny-g**

### CODE REVIEW

File: test/test_client.py [hunk 1]
```diff
@@ -384,17 +384,40 @@ def test_download_aws(self):
...
+        with tempfile.TemporaryDirectory() as d:
+            file_path = utils.download_asset(asset=asset, save_directory=d)
+            with open(file_path, 'rb') as f:
+                data1 = f.read()
+
+            file_path = utils.download_asset(asset=asset, save_filename=file_path)
+            with open(file_path, 'rb') as f:
+                data2 = f.read()
+
+            self.assertEqual(data1, data2)
```
Comment (sunny-g) [inline test/test_client.py]: @davidraleigh had trouble using `assertMultiLineEqual`, got an error about trying to parse the file as a string. Is this right?

Comment (davidraleigh) [reply test/test_client.py]: yup. multiline was for a text file

File: nsl/stac/__init__.py [hunk 2]
```diff
@@ -138,13 +138,17 @@ def authorize(self):
...
+        return self._expiry
+
 
 class AuthGuard:
     def __init__(self, f):
         self.f = f
 
     def __call__(self, *args, **kwargs):
-        if (bearer_auth._expiry - time.time()) < TOKEN_REFRESH_THRESHOLD:
+        if (bearer_auth.expiry - time.time()) < TOKEN_REFRESH_THRESHOLD:
```
Comment (sunny-g) [inline nsl/stac/__init__.py]: :(

**Comment by sunny-g** [comment]
LGTM

**Pull request merged**